sopel/cli/run.py

```diff
@@ -0,0 +1,619 @@
+        stderr('Sopel is not configured, can\'t stop')
+        return ERR_CODE
+
+    # Redirect Outputs
+    utils.redirect_outputs(settings, False)
+
+    # Get Sopel's PID
+    filename = get_pid_filename(opts, settings.core.pid_dir)
+    pid = get_running_pid(filename)
+
+    if pid is None or not tools.check_pid(pid):
+        stderr('Sopel is not running!')
+        return ERR_CODE
+
+    # Stop Sopel
+    if opts.kill:
+        stderr('Killing the Sopel')
+        os.kill(pid, signal.SIGKILL)
+        return
+
+    stderr('Signaling Sopel to stop gracefully')
+    if hasattr(signal, 'SIGUSR1'):
+        os.kill(pid, signal.SIGUSR1)
+    else:
+        # Windows will not generate SIGTERM itself
+        # https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/signal
+        os.kill(pid, signal.SIGTERM)
+
+
+def command_restart(opts):
+    """Restart a running Sopel instance"""
+    # Get Configuration
+    try:
+        settings = utils.load_settings(opts)
+    except ConfigurationNotFound as error:
+        tools.stderr('Configuration "%s" not found' % error.filename)
+        return ERR_CODE
+
+    if settings.core.not_configured:
+        stderr('Sopel is not configured, can\'t stop')
+        return ERR_CODE
+
+    # Redirect Outputs
+    utils.redirect_outputs(settings, False)
+
+    # Get Sopel's PID
+    filename = get_pid_filename(opts, settings.core.pid_dir)
+    pid = get_running_pid(filename)
+
+    if pid is None or not tools.check_pid(pid):
+        stderr('Sopel is not running!')
+        return ERR_CODE
+
+    stderr('Asking Sopel to restart')
+    if hasattr(signal, 'SIGUSR2'):
+        os.kill(pid, signal.SIGUSR2)
+    else:
+        # Windows will not generate SIGILL itself
+        # https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/signal
+        os.kill(pid, signal.SIGILL)
+
+
+def command_legacy(opts):
+    """Legacy Sopel run script
+
+    The ``legacy`` command manages the old-style ``sopel`` command line tool.
+    Most of its features are replaced by the following commands:
+
+    * ``sopel start`` replaces the default behavior (run the bot)
+    * ``sopel stop`` replaces the ``--quit/--kill`` options
+    * ``sopel restart`` replaces the ``--restart`` option
+    * ``sopel configure`` replaces the
+      ``-w/--configure-all/--configure-modules`` options
+
+    The ``-v`` option for "version" is deprecated, ``-V/--version`` should be
+    used instead.
+
+    .. seealso::
+
+       The github issue `#1471`__ tracks various changes requested for future
+       version of Sopel, some of them are related to this legacy command line
+       tool.
+
+       .. __: https://github.com/sopel-irc/sopel/issues/1471
+
+    """
+    # Step One: Handle "No config needed" options
+    if opts.version:
+        print_version()
+        return
+    elif opts.version_legacy:
+        tools.stderr(
+            'option -v is deprecated, '
+            'use `sopel -V/--version` instead')
```

Comment:
All of these deprecation warnings are pretty unobtrusive in actual output. The "error" from the server on quitting is much more conspicuous (because its line actually starts with `ERROR:`… even though it's _not_ an error).

I won't leave line notes on all of them (that would nearly double my line-note count, and be very intimidating when you get this review in your email), but perhaps you could change them all to this format:

```
'WARNING: Option {option_string} is deprecated; '
' use `sopel {replacement_string}` instead!'
```

(Yes, here I'm using a semicolon anyway. :stuck_out_tongue_closed_eyes:)

Reply:
True.